lean4-infoview/src/infoview/rpcSessions.tsx

```diff
@@ -244,9 +249,15 @@ export class RpcSessions implements Disposable {
         })
     }
 
-    dispose() {
+    ensureAllSessionsClosed() {
         this.#connected.forEach(rs => rs.dispose())
         this.#connecting.forEach(fut => fut.then(rs => rs?.dispose()))
```

Comment:
If we are not recreating the `RpcSessions` object, the `then` should include the same lambda as in `ensureSessionClosed`. More simply, `this.#connected.forEach((_, uri) => this.ensureSessionClosed(uri)); this.#connecting.forEach((_, uri) => this.ensureSessionClosed(uri))` should work.